indico/htdocs/js/indico/jquery/categorynavigator.js

```diff
@@ -28,8 +28,22 @@
             openInDialog: false,
             // The title for the category navigator dialog
             dialogTitle: $T.gettext("Select a category"),
-            // Restrict action to categories with no subcategories
-            selectLeafOnly: false,
+            // Allow action to specific categories
+            actionOn: {
+                categoriesWithSubcategories: {
+                    disabled: false,
+                    message: $T.gettext("You cannot select a category that contains subcategories")
+                },
+                categoriesWithEvents: {
+                    disabled: false,
+                    message: $T.gettext("You cannot select a category that contains events")
+                },
+                self: {
+                    disabled: false,
+                    message: $T.gettext("You cannot select the same category"),
```

Comment:
I'm against using **select** on the tooltip default text mostly because the action text is already on the button and won't match if customized. If I see the button saying **Delete** I would be confused if the tooltip says **select**, so to make them match we would be forcing manual customization of the 3 message text options. That's why I'm proposing something more generic. 

It doesn't have to sound technical, so what about `"The action is disabled for this category"`.